docs/src/main/asciidoc/application-configuration-guide.adoc

```diff
@@ -360,6 +360,42 @@ You can override these runtime properties with the following mechanisms (in decr
 
 NOTE: Environment variables names are following the conversion rules of link:https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configsources.asciidoc#default-configsources[Eclipse MicroProfile]
 
+== Using DotEnv files during Development
+As an alternative to defining global environment variables, you can define environment variables in a `.env` file.
+This is practical as it allows each member of your development team to customize and manage the development environment via a file.
+Let's see this in action.
+
+Add the below environment variables in a `src/main/resources/.env` file.
+
+[source,shell]
+--
+GREETING_MESSAGE=hello world
+--
+
+Edit the `application.properties` file so that it looks like:
+
+[source,properties]
+----
+greeting.message = ${GREETING_MESSAGE}
+greeting.name = quarkus
+greeting.hidden.prizeAmount=10
+greeting.hidden.recipients=Jane,John
+----
+
+Now, we will have to tell Quarkus to pick variables defined in the `.env` file with the `-Dquarkus.env.file` system property as shown below.
+[source,shell]
+--
+./mvnw clean compile quarkus:dev -Dquarkus.env.file=.env <1>
+--
+
+1. We start Quarkus application with a path to `.env` file. Absolute path are also accepted.
+
+Open your browser to `http://localhost:8080/greeting` to see the greeting message. Changing the `.env` file will be picked by Quarkus.
+
+NOTE: Variables names in the `.env` file follows the conversion rules of link:https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configsources.asciidoc#default-configsources[Eclipse MicroProfile]
+
+WARNING: DotEnv files are not to be committed in your git repository or to be used in `production` environments (prefer host/global environment variables in this case).
```

Comment:
I am not sure this is absolutely correct. I can easily imagine a scenario where a project would commit an example `.env` file so new developers can easily so which env vars they can override

Reply:
What I have seen in most projects is to ignore a .env file and commit a example.env file.

Reply:
So maybe the documentation should be adapted to take this suggestion into account?

Reply:
+1. Actually thought about it when I was writing my comment. In this case I’ll add it as a tip.

Reply:
@geoand Tip added. WDYT?

Reply:
Looks good!